### PR TITLE
test: add unit tests for core crop math helpers

### PR DESCRIPTION
**Pull request by guoyingtao**
## What

Adds unit-test coverage for three previously **zero-test** pure-function files that carry the highest regression risk in the crop pipeline. These are the ideal things to pin down — no UIKit view state, deterministic math — yet they back features (perspective skew, crop-box resizing, EXIF handling) where a silent numeric error is easy to introduce and hard to notice.

**56 new tests, all passing locally** (`xcodebuild test -scheme Mantis-Package`).

### `PerspectiveTransformHelperTests` (31 tests)
The 3D perspective/skew math behind the v2.29 skew feature:
- `horizontal/verticalSkewTransform3D` — rotation axis (Y vs X), the `m34` perspective term (and how rotation scales it by `cos θ`), the strict `> 10°` translate threshold, translate factor/sign, and sign symmetry.
- `combinedSkewTransform3D` — identity short-circuit, single-axis equivalence with the dedicated helpers, `zoomScale < 1` clamping, `depth / zoom` scaling, and the combined-intensity effective-zoom boost at max skew.
- `projectDisplacement` / `allProjectionsInFrontOfCamera` — perspective division, the degenerate-`w` guard, behind-camera detection.
- `centeredTransform` — corner-count and factor guards, centering shift, factor scaling/clamping.
- `allPointsInsideConvexPolygon` — including the **concave "U"-notch** case that motivated replacing the old cross-product test with ray casting.
- `computeCompensatingScale` — no-scale-needed path and binary-search convergence to the covering scale.

### `GeometryHelperTests` (13 tests)
Crop-box boundary math:
- `getInscribeRect` — same-ratio, width-limited, height-limited, and non-zero origin centering.
- `getCropEdge` — corner priority over adjacent edges (overlap zone), each edge, center/outside → `.none`.
- `scale(from:)` — identity, uniform scale, rotation invariance.

### `UIImageExtensionsTests` (12 tests)
- `exifOrientation` — all **8** orientation → EXIF-integer mappings.
- `cgImageWithFixedOrientation` — dimension swap for left/right families vs. preservation for up/down families.
- `getOutputCropImageSize` — output math plus zero-size / zero-zoom guards.
- `isHorizontal`, `horizontalToVerticalRatio`, `exceedsPixelCount` (incl. the disable and strict-`>` boundary cases).
- `cropWithPerspective` — drives `crop(by:)` with the skew flag set but an **identity warp**, so its output must match the trusted legacy (non-skew) crop pixel-for-pixel, exercising the 100+ line coordinate pipeline end to end.

## Notes
- New files are registered in `Mantis.xcodeproj/project.pbxproj` so the `Mantis-Package` scheme picks them up, following the existing `CICropEquivalenceTests` wiring.
- No production code changed — tests only.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added XCTest coverage for geometry helpers, perspective transform math, and UIImage extension behaviors (orientation handling, cropping, and degenerate/edge cases), including stronger pixel-level equivalence checks.
* **Bug Fixes**
  * Improved system appearance change responsiveness for blur-based visual effects and rotation dial coloring.
* **Other Changes**
  * Updated main-thread constraints for crop view controller delegation and tests.
  * Refined toolbar button inset behavior, plus updated SwiftLint/Xcode metadata and platform deployment targets.

### CODE REVIEW

**Comment by guoyingtao** [comment]
@coderabbitai review

File: Sources/Mantis/CropViewController/CropToolbar.swift [hunk 1]
```diff
@@ -372,7 +372,7 @@ extension CropToolbar {
         button.setContentCompressionResistancePriority(UILayoutPriority(rawValue: compressionPriority), for: .vertical)
 
         button.addTarget(self, action: action, for: .touchUpInside)
-        button.contentEdgeInsets = UIEdgeInsets(top: 4, left: 10, bottom: 4, right: 10)
+        button.configuration?.contentInsets = NSDirectionalEdgeInsets(top: 4, leading: 10, bottom: 4, trailing: 10)
```
[the remark this answers is not in the document]
Comment (guoyingtao) [reply Sources/Mantis/CropViewController/CropToolbar.swift:376]: Fixed
